nrclient/nrclient.go

```diff
@@ -40,26 +45,33 @@ func NewNRClient(cfg config.NRClientConfig, proxyCfg config.ProxyConfig) (*NRCli
 	return nrClient, nil
 }
 
-func (nrClient *NRClient) Send(logRecords []record.LogRecord) (int, error) {
+func (nrClient *NRClient) Send(logRecords []record.LogRecord) (retry bool, err error) {
 	payloads, err := record.PackageRecords(logRecords)
 	if err != nil {
-		return nonRetriableConnectionError, err
+		log.WithField("error", err).Error("Error packaging request")
+		return false, err
 	}
 
 	for _, payload := range payloads {
 		statusCode, err := nrClient.sendPacket(payload)
-		if err != nil || !isSuccesful(statusCode) {
-			return statusCode, err
+		if err != nil {
+			return false, err
 		}
+		if statusCode/100 == 2 {
```

Comment:
I would prefer to have a `if statusCode/100 != 2` and inside it, the logic of returning `true/false, nil` instead the `continue`.

This is just to improve readability of the code and avoid need to read carefully that there is a `continue` and other returns at the end of the for statement will be not executed. 

Mixing this, with the other comment, could be something like:

```
	for _, payload := range payloads {
		statusCode, err := nrClient.sendPacket(payload)
		if err != nil {
			return false, err
		}
		if statusCode/100 != 2 {
			return isStatusCodeRetryable(statusCode), nil
		}
	}
```